l1-contracts/contracts/chain-registrar/ChainRegistrar.sol

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import {IBridgehub} from "../bridgehub/IBridgehub.sol";
+import {PubdataPricingMode} from "../state-transition/chain-deps/ZkSyncHyperchainStorage.sol";
+import {IStateTransitionManager} from "../state-transition/IStateTransitionManager.sol";
+import {ETH_TOKEN_ADDRESS} from "../common/Config.sol";
+import {IERC20} from "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable-v4/access/Ownable2StepUpgradeable.sol";
+import {IGetters} from "../state-transition/chain-interfaces/IGetters.sol";
+import {SafeERC20} from "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
+
+/// @title ChainRegistrar Contract
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @notice This contract is used for proposing and registering new chains in the zkSync ecosystem.
+/// @notice It helps chain administrators retrieve all necessary L1 information about their chain.
+/// @notice Additionally, it assists zkSync administrators in verifying the correctness of registration transactions.
```

Comment:
I don't think "ZKsync administrators" is a good term, also it is a bit confusing as we rather use `ZKsync ecosystem admin` in other places.